pytorch_lightning/trainer/trainer.py

```diff
@@ -1568,7 +1568,7 @@ def training_type_plugin(self) -> TrainingTypePlugin:
 
     @property
     def precision_plugin(self) -> PrecisionPlugin:
-        return self.accelerator.precision_plugin
+        return self.training_type_plugin.precision_plugin
```

Comment:
Related to my comment about it being Optional, if that's the case, then this should check `None` and the typing would need to be fixed. Same for `precision` below

Reply:
It won't be None as 1. in accelerator we make sure precision plugin is not None (there was a resolved conversation about about this) 2. in ttp initialization, if precision_plugin is none, it will auto assign precision_plugin()